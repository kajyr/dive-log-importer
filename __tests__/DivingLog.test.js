const { importer } = require('../src/divelog');
const { parse } = require('../src/xml');
const path = require('path');
const fs = require('fs');

const MOCK_FILE = path.join(__dirname, '__mocks__', 'DivingLog.xml');
const MOCK_DATA = fs.readFileSync(MOCK_FILE, 'utf8');

describe('DiveLog importer', () => {
  test('Basic', async () => {
    const mock = await parse(MOCK_DATA);

    const logbook = importer(mock);
    expect(logbook).toMatchSnapshot();
  });
});
