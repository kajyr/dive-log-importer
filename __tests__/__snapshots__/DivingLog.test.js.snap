// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiveLog importer Basic 1`] = `
Object {
  "dives": Array [
    Object {
      "air_used": 2100,
      "bottom_time": 31.5,
      "buddies": "Andrea, Mauro",
      "computer": "Aladin Sport Matrix",
      "current": "",
      "current_is_calm": true,
      "current_is_strong": false,
      "current_is_weak": false,
      "date": "2017-02-25T10:01:00Z",
      "deco_stops": "-",
      "diveSuit": "umida monopezzo",
      "diveTime": 42,
      "dive_master": "Mauro",
      "emersion_time": 3,
      "entry": "da barca",
      "entry_time": "10:01:00",
      "exit_time": "10:43:00",
      "gear": Array [],
      "half_depth_break": "14m",
      "half_depth_break_time": "2.5",
      "isAir": true,
      "location": Object {
        "country": "Italia",
        "lat": "42.4010",
        "lng": "11.0914",
        "place": "Porto Santo Stefano",
        "site": "Scoglio del Corallo",
      },
      "max_depth": 27.4,
      "notes": "",
      "number": 1,
      "pressure_end": 70,
      "pressure_start": 210,
      "repetitive": true,
      "samples": Array [],
      "surface": "Normale",
      "surfaceInterval": "3:12",
      "surface_is_calm": true,
      "surface_is_mid": false,
      "surface_is_rough": false,
      "type": "addestramento",
      "visibility": "",
      "visibility_is_enough": false,
      "visibility_is_good": true,
      "visibility_is_poor": false,
      "volume_start": 3150,
      "volume_tank": 15,
      "water": "mare",
      "weather": "Sereno",
      "weather_is_clear": true,
      "weather_is_cloud": false,
      "weather_is_rain": false,
      "weights": 9,
    },
  ],
}
`;
